contrib/docker/multi-stage-with-bundle/Dockerfile

```diff
@@ -0,0 +1,42 @@
+# Stage 1 - Create yarn install skeleton layer
+FROM node:14-buster AS packages
+
+WORKDIR /app
+COPY package.json yarn.lock ./
+
+# Uncomment this line if building a non create-app version
+# COPY packages packages
+COPY plugins plugins
+
+RUN find packages \! -name "package.json" -mindepth 2 -maxdepth 2 -print | xargs rm -rf
+
+# Stage 2 - Install dependencies and build packages
+FROM node:14-buster AS build
+
+WORKDIR /app
+COPY --from=packages /app .
+
+RUN yarn install --network-timeout 600000 && rm -rf "$(yarn cache dir)"
+
+COPY . .
+
+RUN yarn tsc
+RUN yarn --cwd packages/backend backstage-cli backend:bundle --build-dependencies
+
+# Stage 3 - Build the actual backend image and install production dependencies
+FROM node:14-buster
+
+WORKDIR /app
+
+# Copy from build stage
+COPY --from=build /app/yarn.lock /app/package.json /app/packages/backend/dist/skeleton.tar.gz ./
+RUN tar xzf skeleton.tar.gz && rm skeleton.tar.gz
+
+RUN yarn install --production --network-timeout 600000 && rm -rf "$(yarn cache dir)"
+
+COPY --from=build /app/packages/backend/dist/bundle.tar.gz .
+RUN tar xzf bundle.tar.gz && rm bundle.tar.gz
+
+COPY app-config.yaml app-config.production.yaml ./
+
+CMD ["node", "packages/backend", "--config", "app-config.yaml", "--config", "app-config.production.yaml"]
```

Comment:
Enable new line on save in [VSCode](https://stackoverflow.com/a/44704969/4698026). :)